homework/index.js

```diff
@@ -30,17 +34,212 @@
+              },
+            },
+          ],
+        },
+      ],
+    },
+  ];
+
+  // Creates html elements from an object
+  function createStaticComponentsFromModel(model, root) {
+    model.forEach(parent => {
+      const parentComponent = createAndAppend(parent.tag, root, parent.attributes);
+
+      if (parent.children) {
+        createStaticComponentsFromModel(parent.children, parentComponent);
+      }
+    });
+  }
+
+  function fillRepositoryDetailsTable(selectedRepository) {
+    const tableBody = document.querySelector('#table-body');
+
+    const tableContent = [
+      ['Repository', 'name'],
+      ['Description', 'description'],
+      ['Forks', 'forks'],
+      ['Updated', 'updated_at'],
+    ];
+
+    tableContent.forEach((item, index) => {
+      const tableRow = createAndAppend('tr', tableBody);
+
+      if (index === 0) {
+        createAndAppend('td', tableRow, { class: 'label', text: item[0] });
+        const td = createAndAppend('td', tableRow);
+        createAndAppend('a', td, {
+          text: selectedRepository[item[1]],
+          href: selectedRepository.html_url,
+          target: '_blank',
+        });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        createAndAppend('td', tableRow, { class: 'label', text: item[0] });
+        createAndAppend('td', tableRow, {
+          text: selectedRepository[item[1]]
+            ? selectedRepository[item[1]]
+            : `No ${item[0].toLowerCase()}`,
+        });
+      }
+    });
+  }
+
+  function fillContributorsList(selectedRepository) {
+    const list = document.querySelector('.contributors__list');
+    fetchJSON(selectedRepository.contributors_url).then(contributors => {
+      if (!contributors.length) {
+        createAndAppend('li', list, {
+          class: 'alert-warning',
+          text: 'There is no contributor for this repository!',
+        });
+        return;
       }
+
+      contributors.forEach(contributor => {
+        const listItem = createAndAppend('li', list, {
+          class: 'contributors__item',
+        });
+        createAndAppend('img', listItem, {
+          class: 'contributors__avatar',
+          src: contributor.avatar_url,
+          alt: contributor.login,
+        });
+        createAndAppend('span', listItem, { class: 'contributors__name', text: contributor.login });
+        createAndAppend('span', listItem, {
+          class: 'contributors__badge',
+          text: contributor.contributions,
+        });
+
+        listItem.addEventListener('click', () => window.open(contributor.html_url));
+      });
     });
   }
 
+  // Starts all actions
+  function init(data, root) {
+    createStaticComponentsFromModel(componentModel, root);
+
+    const repositories = data.sort((a, b) =>
+      a.name.toLowerCase().localeCompare(b.name.toLowerCase()),
+    );
+
+    const select = document.querySelector('.header__select');
+
+    (function listAllRepositoriesIntoSelect() {
```

Comment:
As I commented in week 1, these immediately invoked functions serve no purpose. I can comment out the function heading and closing parts without affecting the result. It is just confusing to other people trying to understand your code and your reasoning.

```js
// (function listAllRepositoriesIntoSelect() {
repositories.forEach((repository, index) => {
  createAndAppend('option', select, { text: repository.name, value: index });
});

const selectedRepository = repositories[select.selectedIndex];

select.addEventListener('change', () => {
  document.querySelector('tbody').innerHTML = '';
  document.querySelector('.contributors__list').innerHTML = '';
  fillRepositoryDetailsTable(selectedRepository);
  fillContributorsList(selectedRepository);
});
// })();

// Shows content randomly when the page is initialized
// (function initializePage() {
const initialRepositoryIndex = Math.floor(Math.random() * repositories.length);

select.selectedIndex = initialRepositoryIndex;
fillRepositoryDetailsTable(repositories[initialRepositoryIndex]);
fillContributorsList(repositories[initialRepositoryIndex]);
// })();
```